extension/src/plots/model/collect.ts

```diff
@@ -235,38 +247,47 @@ export const collectCheckpointPlotsData = (
     }
   }
 
+  const plotsData: CustomCheckpointPlots = {}
   if (acc.plots.size === 0) {
-    return
+    return plotsData
   }
 
-  const plotsData: CheckpointPlot[] = []
-
   for (const [key, value] of acc.plots.entries()) {
-    plotsData.push({ id: decodeColumn(key), values: value })
+    const decodedMetric = decodeColumn(key)
+    plotsData[decodedMetric] = {
+      id: getCustomPlotId(decodedMetric),
+      metric: decodedMetric,
+      param: CHECKPOINTS_PARAM,
+      type: CustomPlotType.CHECKPOINT,
+      values: value
+    }
   }
 
   return plotsData
 }
 
-export const getCustomPlotId = (metric: string, param: string) =>
-  `custom-${metric}-${param}`
-
-const collectCustomPlotData = (
+const collectMetricVsParamPlot = (
```

Comment:
Apologies, I'm a little confused by your question. Both of those functions do very different things at the moment (one relies on output data while other relies on experiment data)... Though we could definitely look into consolidating how we get the custom plot data. I think we could gather checkpoint data with experiment data possibly 🤔

Reply:
I mean I don't understand why every time there is something done to custom plots it has to be separated between `checkpoint` and `metricVsParam` when most of the time the difference between could be handle by the same function the type being one or the other. We don't actually have to make the distinction between both in most places.

Reply:
I would rather not introduce a temporary data structure to be replaced later. This usually creates tech debt and here the separation introduces complexity that we'll have to deal with until it is fixed.

Reply:
When you say temporary data structure, are you referring to how the checkpoints plots are being held in an object?

Reply:
> I would rather not introduce a temporary data structure to be replaced later. This usually creates tech debt and here the separation introduces complexity that we'll have to deal with until it is fixed.

I understand not wanting to create a tech debt, but I really think we need to get this merged. But, the longer we wait the harder it's going to be to merge it because it alters so many different plot files. What do you think?

Reply:
I get it, it works as expected when we run the extension and there will be a lot of conflicts, but we shouldn't rush to merge things because of this unless it fixes a bug. Most of the conflicts you'll encounter are going to be because we created a temporary duplication before (custom and checkpoints). Creating temporary data structures will simply make more merge conflicts later. This PR is about moving trend plots inside custom plots, why keep treating them separately then? In other words, I feel like this is the heart of the PR. I see it more as a fix inside this PR concern than a follow up. If everyone else is fine merging this, then I'll let go.

Reply:
Makes sense... @mattseddon what do you think?

If Matt agrees, I'll go ahead and convert this into a draft and work on a second pr that has both checkpoint/custom data gathered with experiment data. That should take care of the temporary data structure (`customCheckpoints`) we have.

Reply:
I think we can delay merging in order to get this right.

A few takeaways for me and to discuss in the retro:

1. We probably should not have merged https://github.com/iterative/vscode-dvc/pull/3342.
2. Best way forwards after https://github.com/iterative/vscode-dvc/pull/3342 may have been to chain off a revert.
3. When PRs are this big we need to shorten the feedback loop (pair/mob is best option).
4. We need to fight against making so many changes in a single PR. Find some other way to discuss data structures/approaches early and avoid thrashing.

🙏🏻